File: dp_labs_and_solutions/dp_labs_and_solutions/dp_matlab_labs_and_solutions/mediator/solution/SetupClass.m
% SetupClass.m

classdef SetupClass < handle
	methods (Static)
		function result = setup
			aProducer1 = Producer('Alpha');
			aProducer2 = Producer('Bravo');
			aProducer3 = Producer('Charlie');
			aProducer4 = Producer('Delta');
			aProducer5 = Producer('Echo');
			aProducer6 = Producer('Foxtrot');

			aMiddleman = Middleman;
			aMiddleman.addProducer(aProducer1);
			aMiddleman.addProducer(aProducer2);
			aMiddleman.addProducer(aProducer3);
			aMiddleman.addProducer(aProducer4);
			aMiddleman.addProducer(aProducer5);
			aMiddleman.addProducer(aProducer6);

			aConsumer = Consumer('Zulu', aMiddleman);

			result = aConsumer;
		end
	end
end

File: dp_labs_and_solutions/original/dp_labs_and_solutions/dp_cpp_labs_and_solutions/mediator/lab/setup_class.cpp
// setup_class.cpp

#include "consumer.h"
#include "producer.h"
#include "setup_class.h"

consumer * setup_class::setup()
{
	producer * a_producer1 = new producer("Alpha");
	producer * a_producer2 = new producer("Bravo");
	producer * a_producer3 = new producer("Charlie");
	producer * a_producer4 = new producer("Delta");
	producer * a_producer5 = new producer("Echo");
	producer * a_producer6 = new producer("Foxtrot");

	consumer * a_consumer = new consumer("Zulu");
	a_consumer->add_producer(a_producer1);
	a_consumer->add_producer(a_producer2);
	a_consumer->add_producer(a_producer3);
	a_consumer->add_producer(a_producer4);
	a_consumer->add_producer(a_producer5);
	a_consumer->add_producer(a_producer6);

	return a_consumer;
}
